### PR TITLE
Setup Travis CI

### PR DESCRIPTION
**Pull request by jhump**
The travis config and `Makefile` are basically copied from [grpcui](https://github.com/fullstorydev/grpcui).

### CODE REVIEW

**Comment by jhump** [comment]
@jameremo, this isn't yet ready to review. I based it on your branch, #36, to make sure I can get Travis stuff to pass. So I'll rebase after your PR lands (and after I get Travis to actually pass).

**Comment by jhump** [comment]
Oh, it's merged. That was fast. Will fix this up soon.

**Comment by jameremo** [comment]
I want to get this repo into a good state as soon as possible, I'm frankly embarrassed it hasn't been buildable for over a month and we didn't know. Your efforts in that direction are so so appreciated.

**Comment by jhump** [comment]
@jameremo, okay, looks like I got it passing. I went ahead and update this repo's branch settings, so that changes must pass Travis before they can be merged to master.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+language: go
+sudo: false
+
+matrix:
+  include:
+    - go: "1.9"
```
[the remark this answers is not in the document]
Comment (jhump) [reply .travis.yml]: I went ahead and added 1.8 and 1.7, to see if they work. But a lot of 3rd party libraries have stopped supporting 1.8 and older. So if travis fails for either of those versions, I'll just change the README instead.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	predeclared $(all_to_check)
+
+# Intentionally omitted from CI, but target here for ad-hoc reports.
+.PHONY: golint
+golint:
+	@go get golang.org/x/lint/golint
+	golint -min_confidence 0.9 -set_exit_status . $(dirs_to_check)
+
+# Intentionally omitted from CI, but target here for ad-hoc reports.
+.PHONY: errchack
```
[the remark this answers is not in the document]
Comment (jhump) [reply Makefile]: Doh! Nope. And it's been copy-pasta'ed all over the place -- so this typo can be found in many of our repos :(

Fixed. (And sending out PRs to fix in some other repos, too...)

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhump) [reply .travis.yml]: Looks like we have some dependencies that can no longer compiler under Go 1.7. I'll update the README.

**Comment by jhump** [comment]
FYI: The main difference between this and the Makefiles in our other repos, such as `grpcurl`, is that this one does not try to do a `go get -t ./...`, to download and install all dependencies. That is because hauser is a stand-alone command-line tool, not a library, so it vendors all of its dependencies. So it needs to build successfully in an empty GOPATH, without having done `go get` for anything.

**Pull request merged**